### PR TITLE
feat(FI-858) test for deduplication

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR adds a deduplication test for the ICRC-1 test suite

### CODE REVIEW

File: test/suite/lib.rs [hunk 1]
```diff
@@ -218,13 +219,80 @@ pub async fn test_supported_standards(ledger: impl LedgerEnv) -> anyhow::Result<
...
+            < transfer(&p1_env, transfer_args.clone())
+                .await
+                .unwrap()
+                .unwrap()
+    );
+
+    let now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_nanos();
```
Comment (roman-kashitsyn) [inline test/suite/lib.rs:244]: That code might not work with the StateMachine tests.
We need `current_time()` or something like that in LedgerEnv.

File: test/suite/lib.rs [hunk 2]
```diff
@@ -218,13 +219,80 @@ pub async fn test_supported_standards(ledger: impl LedgerEnv) -> anyhow::Result<
...
+    assert!(
+        transfer(&p1_env, transfer_args.clone())
+            .await
+            .unwrap()
+            .unwrap()
+            < transfer(&p1_env, transfer_args.clone())
+                .await
+                .unwrap()
+                .unwrap()
+    );
```
Comment (roman-kashitsyn) [inline test/suite/lib.rs:239]: We must not panic in the test suite because this will crash the entire test process. We should return an `Outcome` instead.

File: test/suite/lib.rs (hunk 1 above)
Comment (NikolasHai) [reply test/suite/lib.rs:244]: You were right, I fixed it

File: test/suite/lib.rs [hunk 3]
```diff
@@ -218,13 +219,80 @@ pub async fn test_supported_standards(ledger: impl LedgerEnv) -> anyhow::Result<
     Ok(Outcome::Passed)
 }
 
+/// Checks whether the ledger advertizes support for ICRC-1 standard.
```
Comment (maciejdfinity) [inline test/suite/lib.rs:222]: comment copy paste :)

File: test/suite/lib.rs [hunk 4]
```diff
@@ -218,13 +219,84 @@ pub async fn test_supported_standards(ledger: impl LedgerEnv) -> anyhow::Result<
...
+        .await
+        .unwrap()
+        .unwrap();
+
+    // If we send the transfer again it should be a duplicate --> Transfer should not succeed
+    assert!(transfer(&p1_env, transfer_args.clone())
+        .await
+        .unwrap()
+        .is_err());
+
```
Comment (maciejdfinity) [inline test/suite/lib.rs:285]: We could also test whether the ledger correctly identifies that transfer to (principal, None) and (principal, Some([0; 32])) are different transactions and are not deduplicated.

Comment (NikolasHai) [reply test/suite/lib.rs:285]: Good idea

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -1 +1 @@
-@dfinity/finint
+@dfinity/finint
```
Comment (roman-kashitsyn) [inline .github/CODEOWNERS:1]: Why did we change this file?

**Review by roman-kashitsyn — APPROVED** [review]
LGTM, thanks!
